analytics_data_api/v0/urls/engagement_timelines.py

```diff
@@ -0,0 +1,14 @@
+from django.conf.urls import patterns, url
+
+from analytics_data_api.v0.views import engagement_timelines as views
+from analytics_data_api.v0.urls import USERNAME_PATTERN
+
+ENGAGEMENT_TIMELINE_URLS = [
+    ('', views.EngagementTimelineView, 'engagement_timelines')
+]
+
+urlpatterns = []
+
+for path, view, name in ENGAGEMENT_TIMELINE_URLS:
+    regex = r'^{0}/$'.format(USERNAME_PATTERN)
+    urlpatterns += patterns('', url(regex, view.as_view(), name=name))
```

Comment:
I'm not a huge fan of this pattern for constructing URLs - I find that at a quick glance it's hard to tell what the URLs in this file are, especially since there's just a single url, which is r'^(?P<username>.+)/$'. The pattern I'm used to is as follows (the example is `urls/learners.py` taken from my learner list PR):

```python
from django.conf.urls import patterns, url

from analytics_data_api.v0.views import learners as views


USERNAME_PATTERN = r'(?P<username>.+)'

urlpatterns = patterns(
    '',
    url(r'^$', views.LearnerListView.as_view(), name='learners'),
    url(r'^{}/$'.format(USERNAME_PATTERN), views.LearnerView.as_view(), name='learner'),
)
```